cmd/operator-opamp-bridge/config/config.go

```diff
@@ -16,43 +16,99 @@ package config
 
 import (
 	"crypto/rand"
+	"errors"
 	"fmt"
+	"io/fs"
+	"net/url"
 	"os"
 	"runtime"
 	"time"
 
+	"github.com/go-logr/logr"
 	"github.com/oklog/ulid/v2"
-	"github.com/open-telemetry/opamp-go/client"
+	opampclient "github.com/open-telemetry/opamp-go/client"
 	"github.com/open-telemetry/opamp-go/protobufs"
+	"github.com/spf13/pflag"
 	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
 	"github.com/open-telemetry/opentelemetry-operator/cmd/operator-opamp-bridge/logger"
 )
 
 const (
-	agentType             = "io.opentelemetry.operator-opamp-bridge"
-	defaultConfigFilePath = "/conf/remoteconfiguration.yaml"
+	agentType = "io.opentelemetry.operator-opamp-bridge"
 )
 
 var (
-	agentVersion = os.Getenv("OPAMP_VERSION")
-	hostname, _  = os.Hostname()
+	agentVersion  = os.Getenv("OPAMP_VERSION")
+	hostname, _   = os.Hostname()
+	schemeBuilder = k8sruntime.NewSchemeBuilder(registerKnownTypes)
+)
+
+func registerKnownTypes(s *k8sruntime.Scheme) error {
+	s.AddKnownTypes(v1alpha1.GroupVersion, &v1alpha1.OpenTelemetryCollector{}, &v1alpha1.OpenTelemetryCollectorList{})
+	metav1.AddToGroupVersion(s, v1alpha1.GroupVersion)
+	return nil
+}
+
+func GetLogger() logr.Logger {
+	return zap.New(zap.UseFlagOptions(&zapCmdLineOpts))
+}
+
+type Capability string
+
+const (
+	Unspecified                    Capability = "Unspecified"
+	ReportsStatus                  Capability = "ReportsStatus"
+	AcceptsRemoteConfig            Capability = "AcceptsRemoteConfig"
+	ReportsEffectiveConfig         Capability = "ReportsEffectiveConfig"
+	AcceptsPackages                Capability = "AcceptsPackages"
+	ReportsPackageStatuses         Capability = "ReportsPackageStatuses"
+	ReportsOwnTraces               Capability = "ReportsOwnTraces"
+	ReportsOwnMetrics              Capability = "ReportsOwnMetrics"
+	ReportsOwnLogs                 Capability = "ReportsOwnLogs"
+	AcceptsOpAMPConnectionSettings Capability = "AcceptsOpAMPConnectionSettings"
+	AcceptsOtherConnectionSettings Capability = "AcceptsOtherConnectionSettings"
+	AcceptsRestartCommand          Capability = "AcceptsRestartCommand"
+	ReportsHealth                  Capability = "ReportsHealth"
+	ReportsRemoteConfig            Capability = "ReportsRemoteConfig"
 )
 
 type Config struct {
-	Endpoint     string   `yaml:"endpoint"`
-	Protocol     string   `yaml:"protocol"`
-	Capabilities []string `yaml:"capabilities"`
+	// KubeConfigFilePath empty if in cluster configuration is in use
```

Comment:
Something about this wording is a little confusing, maybe you have an idea of how to clarify. Or maybe there's an extra "in" in here

Reply:
yeah i can clear that up